src/replication.c

```diff
@@ -3454,6 +3457,20 @@ int replicationCountAcksByOffset(long long offset) {
     return count;
 }
 
+void waitForReplication(client *c, mstime_t timeout, long numreplicas, long long offset, int btype) {
```

Comment:
So, I'm still a bit uneasy about having two distinct systems for managing slots, but I really like the idea of having the replication stream be the authoritative state of slots. I.E. Replicas determine which slots they own from their replication stream, not from the clusterbus state. I'd like to think through this more to decide if there are more changes we need here.